tensorboard/plugins/mesh/tf_mesh_dashboard/mesh-loader.js

```diff
@@ -163,15 +159,54 @@ Polymer({
 
   /**
    * Updates the scene.
-   * @param {!Object} currentStep Step datum.
    * @private
    */
-  _updateScene: function(currentStep) {
+  _updateScene: function() {
+    const currentStep = this._currentStep;
+    // Mesh data is not fetched yet. Please see `_maybeFetchMesh`.
+    if (!currentStep.mesh) return;
+
     this._meshViewer.updateScene(currentStep, this);
     if (!this._cameraPositionInitialized) {
       this._meshViewer.resetView();
       this._cameraPositionInitialized = true;
     }
+    if (!this._meshViewerAttached) {
+      // Mesh viewer should be added to the dom once.
+      this.appendChild(this._meshViewer.getRenderer().domElement);
+      this._meshViewerAttached = true;
+    }
+  },
+
+  _debouncedFetchMesh() {
+    this.debounce('fetchMesh', () => this._maybeFetchMesh(), 100);
+  },
+
+  async _maybeFetchMesh() {
+    const currentStep = this._currentStep;
+    if (currentStep.mesh || currentStep.meshFetching) return;
+    currentStep.meshFetching = true;
+    this._isMeshLoading = true;
```

Comment:
Just letting you know: do note that this can lead to incorrect UI state. 
Test case:
1. go to the network tab in Chrome and enable throttling to emulate slow internet
2. move the step slider with fervor

Multiple requests should be in flight at the same time and `this._isMeshLoading` will be `false` the moment one of them resolve the first even while others are in-flight.

Reply:
Thanks, created ticket.